packages/firestore/test/integration/api/database.test.ts

```diff
@@ -925,13 +926,29 @@ apiDescribe('Database', (persistence: boolean) => {
 
   it('exposes "firestore" on document references.', () => {
     return withTestDb(persistence, async db => {
-      expect(db.doc('foo/bar').firestore).to.equal(db);
+      if (usesModularApi()) {
+        // FirestoreShim returns a new instance of FirebaseFirestore that is
+        // semantically identical to the originating instance, but does not
+        // compare equal using the default JavaScript semantics.
+        expect(db.doc('foo/bar').firestore).to.be.an.instanceof(Firestore);
```

Comment:
Is there a way we could define equality of Firestore instances based on e.g. the project, database, and app name?

I don't mean necessarily adding equals methods or something like that, but just an assertion we could use in tests? That way we could assert the Firestore instances are equivalent and use the same assertion for both cases.

As it stands, verifying that the thing returned is an instance of Firestore ... isn't much of a test.

Reply:
I changed the shim to pass the Firestore instance between types, which is likely what we need to to once we productize this code path. This change removes the need to special-case this here.